### PR TITLE
Lasb 3580/get reporder by repid

### PR DESCRIPTION
**Pull request by Ross-Nation**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-3580)

Describe what you did and why.

Added an endpoint to the RepOrderController to return rep order state details. Tested locally with postman and all running as expected. 

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/controller/RepOrderControllerTest.java [hunk 1]
```diff
@@ -410,6 +390,46 @@ void givenValidUsn_whenFindRepOrderByUsnIsInvokedAndFindsNoRepOrder_thenCorrectR
...
+            .andExpect(jsonPath("$.caseType").value(CASE_TYPE_VALUE))
+            .andExpect(jsonPath("$.iojResult").value(IOJ_RESULT_VALUE))
+            .andExpect(jsonPath("$.iojAssessorName.fullName").value(IOJ_ASSESSOR_FULL_NAME))
+            .andExpect(jsonPath("$.iojAssessorName.userName").value(IOJ_ASSESSOR_USERNAME))
+            .andExpect(jsonPath("$.dateAppCreated").value(DATE_APP_CREATED_VALUE))
+            .andExpect(jsonPath("$.iojReason").doesNotExist())
+            .andExpect(jsonPath("$.meansInitResult").value(MEANS_INIT_RESULT_VALUE))
+            .andExpect(jsonPath("$.meansInitStatus").value(MEANS_INIT_STATUS_VALUE))
+            .andExpect(jsonPath("$.meansFullResult").doesNotExist())
+            .andExpect(jsonPath("$.meansFullStatus").doesNotExist())
```
Comment (JHunt16) [inline maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/controller/RepOrderControllerTest.java:413]: Do we need a test for the state of a full means assessment?

Comment (Ross-Nation) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/controller/RepOrderControllerTest.java:413]: Good question. I'm ngl, I ripped this off of the USN endpoint tests so maybe didn't think enough about it. Let me have a look. 

I know the mapping of all fields is tested in the mapper tests and I did also test this manually so I know it works... but I will think about it and get back to you!

Comment (Ross-Nation) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/controller/RepOrderControllerTest.java:413]: So I had a look at maybe creating a few different mapping scenarios to cover each of the fields but looking at the mapping tests, this is already done there. So I'm leaning on the side of this being unnecessary duplication to do it in the controller tests as well. 

Mapper tests are here: https://github.com/ministryofjustice/laa-maat-court-data-api/blob/main/maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/mapper/RepOrderMapperTest.java

Comment (JHunt16) [reply maat-court-data-api/src/test/java/gov/uk/courtdata/reporder/controller/RepOrderControllerTest.java:413]: Makes sense, I agree no need to duplicate as the mapping is already tested.

**Review by JHunt16 — APPROVED** [review]

**Pull request merged**